Implementing automatic htaccess generation (#68)
ai-robots-txt/ai.robots.txt: README.md

## Usage

Many visitors will find these files from this repository most useful:
- `robots.txt`
- `.htaccess`

`robots.txt` implements the Robots Exclusion Protocol ([RFC 9309](https://www.rfc-editor.org/rfc/rfc9309.html)).

The second one tells your own webserver to return an error page when one of the listed AI crawlers tries to request a page from your website. A `.htaccess` file does not work on every webserver, but works correctly on most common and cheap shared hosting providers. The majority of AI crawlers set a "User Agent" string in every request they send, by which they are identifiable: this string is used to filter the request. Instead of simply hoping the crawler pledges to respect our intention, this solution actively sends back a bad webpage (an error or an empty page). Note that this solution isn't bulletproof either, as anyone can fake the sent User Agent.

Note that, as stated in the [httpd documentation](https://httpd.apache.org/docs/current/howto/htaccess.html), more performant methods than an `.htaccess` file exist. Nevertheless, most shared hosting providers only allow `.htaccess` configuration.

## Contributing

> A note about contributing: updates should be added/made to `robots.json`. A GitHub action, courtesy of [Adam](https://github.com/newbold), will then generate the updated `robots.txt` and `table-of-bot-metrics.md`.

Reviewer: ```suggestion
A note about contributing: updates should be added/made to `robots.json`. A GitHub action will then generate the updated `robots.txt`, `table-of-bot-metrics.md`, and `.htaccess`.
```

If you'd like to add information about a crawler to the list, please make a pull request with the bot name added to `robots.txt`, `ai.txt`, and any relevant details in `table-of-bot-metrics.md` to help people understand what's crawling.

## Usage

> Many visitors will find these files from this repository most useful:

Reviewer: Who are the "visitors"? I suggest adopting a more formal tone and breaking out subsections for `robots.txt` and `.htaccess`.

> Many visitors will find these files from this repository most useful:
> - `robots.txt`
> - `.htaccess`

Reviewer: ```suggestion
This repository provides the following files:
- `robots.txt`
- `.htaccess`

### `robots.txt`
```

> The first one tells search engine and AI crawlers which parts of your website should be scanned or avoided. The webpages of your server are returned anyway, but the crawler "pledges" not to use them. By default, the provided `robots.txt` tells every AI crawler not to scan any page in your website. This is not bulletproof, as an evil crawler could simply ignore the `robots.txt` content.

Reviewer: I suggest keeping the description of `robots.txt` minimal in this PR. Others can fill it out later.
```suggestion
`robots.txt` implements the Robots Exclusion Protocol ([RFC 9309](https://www.rfc-editor.org/rfc/rfc9309.html)).
```

> The second one tells your own webserver to return an error page when one of the listed AI crawlers tries to request a page from your website. A `.htaccess` file does not work on every webserver, but works correctly on most common and cheap shared hosting providers. The majority of AI crawlers set a "User Agent" string in every request they send, by which they are identifiable: this string is used to filter the request. Instead of simply hoping the crawler pledges to respect our intention, this solution actively sends back a bad webpage (an error or an empty page). Note that this solution isn't bulletproof either, as anyone can fake the sent User Agent.

Reviewer: I suggest trimming this back to something basic for this PR. Others can fill out the details later.
```suggestion
### `.htaccess`

`.htaccess` may be used to configure web servers such as [Apache HTTPD](https://httpd.apache.org/) to return an error page when one of the listed AI crawlers sends a request to the web server.
```

> Note that, as stated in the [httpd documentation](https://httpd.apache.org/docs/current/howto/htaccess.html), more performant methods than an `.htaccess` file exist. Nevertheless, most shared hosting providers only allow `.htaccess` configuration.

Reviewer: Let's keep the statement very simple for now:
```suggestion
Note that, as stated in the [httpd documentation](https://httpd.apache.org/docs/current/howto/htaccess.html), more performant methods than an `.htaccess` file exist.
```

> We suggest adding both files, as some crawlers may respect `robots.txt` while not having an identifiable User Agent; on the other hand, other crawlers may not respect the `robots.txt`, but they provide a identifiable User Agent by which we can filter them out.

Reviewer: I don't understand the statement "some crawlers may respect `robots.txt` while not having an identifiable User Agent" because our `robots.txt` is nothing more than a list of User Agents for which access to files is denied. So, unless a crawler has a User Agent in the list, `robots.txt` will have no effect.

> We suggest adding both files, as some crawlers may respect `robots.txt` while not having an identifiable User Agent; on the other hand, other crawlers may not respect the `robots.txt`, but they provide a identifiable User Agent by which we can filter them out.

Reviewer: I think it's safest to avoid this sentence altogether as the content is hard to follow.
```suggestion
```

## Contributing

> A note about contributing: updates should be added/made to `robots.json`. A GitHub action, courtesy of [Adam](https://github.com/newbold), will then generate the updated `robots.txt` and `table-of-bot-metrics.md`.

Reviewer: For consistency, this should mention that `.htaccess` is also generated.

Author: Should be solved in https://github.com/ai-robots-txt/ai.robots.txt/pull/68/commits/70fd6c0fb13cdf4f0525bf061556e8e50ca7b8d9 as in your suggestion in https://github.com/ai-robots-txt/ai.robots.txt/pull/68#discussion_r1921072354
